kof-operator/internal/k8s/container.go

```diff
@@ -0,0 +1,39 @@
+package k8s
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func GetContainer(containers []corev1.Container, name string) (*corev1.Container, error) {
+	for _, container := range containers {
+		if container.Name == name {
+			return &container, nil
+		}
+	}
+	return nil, fmt.Errorf("container %s not found in spec", name)
```

Comment:
This error is not needed, as the clients also raises an error if container not found. This is kinda a duplication, but as it does no harm you might keep it.